src/effect/delay.rs

```diff
@@ -54,15 +58,22 @@ impl Delay {
         }
     }
 
-    fn handle_midi_messages(&mut self, messages: &[MidiMessage<'static>]) -> Result<()> {
+    fn handle_midi_messages(&mut self, messages: &[midi::Message]) -> Result<()> {
         if let Some(delay) = self.delay_from_midi_messages(messages) {
             self.set_delay(delay)?;
         }
 
         Ok(())
     }
 
-    fn delay_from_midi_messages(&self, messages: &[MidiMessage<'static>]) -> Option<u32> {
+    fn delay_from_midi_messages(&mut self, messages: &[midi::Message]) -> Option<u32> {
+        let delay_from_slider = self.delay_from_midi_messages_slider(messages);
+        let delay_from_tap_tempo = self.delay_from_midi_messages_tap(messages);
```

Comment:
Always check both, even though we prefer the delay from the slider (arbitrarily), since `TapTempo` is a state machine and needs to see all the MIDI messages that come in.